driver-core/src/main/java/com/datastax/driver/core/PagingState.java

```diff
@@ -85,7 +85,8 @@ private PagingState(byte[] complete) {
                 //it is a RegularStatement since Batch statements are not allowed
                 RegularStatement rs = (RegularStatement)statement;
                 md.update(rs.getQueryString().getBytes());
-                values = rs.getValues(protocolVersion);
+                // FIXME use per-cluster configured CodecRegistry
+                values = rs.getValues(protocolVersion, new CodecRegistry());
```

Comment:
Looking at the callers for ArrayBackedResultSet.fromMessage() it looks like Session is only provided as null if creating a DefaultResultSetFuture from a ControlConnection.  In that case, we have a Cluster object, so we can probably get to the CodecRegistry somehow that way, although not trivial withoug some API changes in DefaultResultSetFuture.

I think PagingState could work if we propagated the CodecRegistry into ExecutionInfo that does seem awkward though.

Since TupleType.of is deprecated we can probably get away with removing it right?